### PR TITLE
Add function in datasets to get input paths

### PR DESCRIPTION
**Pull request by anwai98**
This PR extends support for the possibility of having paths to input images / volumes. This is in favor of https://github.com/computational-cell-analytics/micro-sam/pull/728 to avoid writing additional scripts to extracts paths split-wise / data partition wise.

It's a WIP, I'll put this in review once the required (ideally all) datasets have the support for the mentioned feature.

### CODE REVIEW

File: torch_em/data/datasets/light_microscopy/deepbacs.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def get_deepbacs_data(path: Union[os.PathLike, str], bac_type: str, download: bo
     return data_folder
 
 
-def _get_paths(path, bac_type, split):
+def _get_deepbacs_paths(path, bac_type, split):
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/deepbacs.py:103]: If we make this a standard across dataset then we can remove the `_` and add docstrings.

**Review by constantinpape — COMMENTED** [review]
I think this is a good idea. If we standardize it we can also document it though.

File: torch_em/data/datasets/light_microscopy/deepbacs.py (hunk 1 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/deepbacs.py]: Thanks for the suggestion. I decided to go ahead and implement this for all microscopy datasets now.

**Review by constantinpape — APPROVED** [review]
Looks good high-level. I didn't check all changes in detail. We'll find issues when using the datasets.

**Pull request merged**